### DISCUSSION TITLE
Refraction Implementation

### DISCUSSION THREAD

**Discussion by Anplus** [NVlabs/sionna#528] [General]
I am working on implementing the refraction model detailed in the ITU’s ‘Effects of building materials and structures on radiowave propagation above about 100 MHz,’ focusing on the single slab model with internal reflections. My approach modifies the Fibonacci method to trace possible first refraction paths and logs the incident points along with the slab thickness at these points. Subsequently, the field computation will be adjusted to calculate the transmission and reflection coefficients based on the ITU model.

I am reaching out to inquire if this enhancement aligns with Sionna’s current development roadmap. Could you please let me know if there is already similar work underway or planned within your projects?

**Comment by jhoydis** [comment]
Hi @Anplus,

Thanks for reaching out. We have indeed an already ongoing effort targeting refraction.
However, we cannot comment on the release date for this feature.

**Reply by JackYifan** [reply]
Thank you for your efforts, we're really looking forward to this feature!

**Reply by HelmutWolfgang** [reply]
Hello @jhoydis,

can you provide an update regarding the progress or even the release date of refraction?

Thank you.

**Reply by jhoydis** [reply]
It will be included in the next release. I cannot comment on the timeline.

**Reply by merlinND** [reply]
Sionna 1.0.1 has been release with built-in support for refraction.
